Automattic/wp-calypso
# Update Me: 2FA backup codes list to have to ability to save codes to disk

Issue #385

This pull request seeks to add a button to the backup codes list that will allow a user to save their codes to the disk. The filename for the saved codes is currently configured to `[USERNAME]-backup-codes.txt`.

**Implementation notes:**

This just uses the built-in browser `Blob` object and the `browser-filesaver` library to facilitate the saving of the file to disk.

**Testing instructions:**

1. Navigate to the two-step authentication page.
2. Click the Generate New Backup Codes button.
3. Once new codes have been generated click the Save button.
4. Confirm that the codes have been saved to disk successfully.

**Notes:**
While this currently users the built-in `Blob` object there is a limitation with this as there are some older browsers that don't implement it. If it is desired to support older browsers with this implementation the polyfill `blob-polyfill` could be used. I did notice that there is already a `blob` module installed but it looks like it only falls back to `BlobBuilder` if `Blob` cannot be found (and reading up on `BlobBuilder` it looks like it is not considered obsolete) so I'm not sure if that would be a viable option.
